src/chunk-stream.iced

```diff
@@ -20,28 +28,42 @@ exports.ChunkStream = class ChunkStream extends stream.Transform
     # if we don't have a full block, dump everything into extra and skip this round
     if chunk.length < @block_size
       @extra = chunk
-    else
-      # if we have to, call the transform function multiple times until we have less than a block size remaining
-      if @exact_chunking
-        while chunk.length >= @block_size
-          @push(@transform_func(chunk[...@block_size]))
-          chunk = chunk[@block_size...]
-        @extra = chunk
-      # if the transform function can accept chunks of length block_size*n for n \in N, just make one transform function call
-      else
-        remainder = chunk.length % @block_size
-        # chop off extra
-        if remainder isnt 0
-          @extra = chunk[chunk.length-remainder...]
-          chunk = chunk[...chunk.length-remainder]
-        @push(@transform_func(chunk))
-
-    cb()
+      return cb(null, new Buffer(''))
+
+    # next, mangle the chunk so that it is an even multiple of the block size
+    remainder = chunk.length % @block_size
+    if remainder isnt 0
+      @extra = chunk[chunk.length-remainder...]
+      chunk = chunk[...chunk.length-remainder]
+
+    # finally, process the chunk (guaranteed to be the correct length)
+    esc = make_esc(cb, "Failed to transform the chunk")
```

Comment:
No, sorry. What I mean is that the `make_esc` call should probably go to the top of the function and look something like this:

    esc = make_esc(cb, "ChunkStream::_transform")

Then if you decide that errors in `_transform` need to have extra context attached to them (that is, context that will make sense if it's printed in the terminal to a human), it could look something like this:

    await @_transform_chunk(chunk, defer(err, out))
    if err?
        return cb(new Error("Failed to transform the chunk: #{err}"), null)

That said, the chunk stream itself is probably an implementation detail that won't mean much to an end user. It might be enough to just rely on the `esc` stack trace here and return errors unmodified.